### PR TITLE
Swap to adam_core's Propagator, Orbits, and Observers

### PR DESCRIPTION
**Pull request by moeyensj**
Replaces:
- Backend class with adam_core's Propagator class
- Orbits with adam_core's Orbits
- Observers with adam_core's Observers
- coordinate transformations with adam_core's transformations

Removes:
- Old implementation of range and shift
- Data pre-processing function that will not be necessary with the new typed structure of Observations

All unit tests pass, but most importantly:
`pytest -k ivezic -m integration` passes! This tests the entire pipeline.

The benchmark shows a slowdown from 1.2695 to 1.4256 seconds. This is more than likely due to the conversion from adam_core types to dataframes still used by a large part of the THOR pipeline. I'll change those types out in a follow up PR.

### CODE REVIEW

File: thor/orbits/attribution.py [hunk 1]
```diff
@@ -40,30 +46,34 @@ def attribution_worker(
...
     # Genereate ephemerides for each orbit at the observation times
-    ephemeris = generateEphemeris(
-        orbits,
-        observers,
-        backend=backend,
-        backend_kwargs=backend_kwargs,
-        num_jobs=1,
-        chunk_size=1,
-    )
+    ephemeris = prop._generate_ephemeris(orbits, observers)
```
Comment (akoumjian) [inline thor/orbits/attribution.py:74]: Why do we use the private method here?

File: thor/orbits/iod.py [hunk 2]
```diff
@@ -178,8 +184,8 @@ def iod(
     observation_selection_method="combinations",
     iterate=False,
     light_time=True,
-    backend="PYOORB",
-    backend_kwargs={},
+    propagator: Literal["PYOORB"] = "PYOORB",
```
Comment (akoumjian) [inline thor/orbits/iod.py:187]: Let's think about how to make this open enough that people can install third-party packages and use other propagators.

**Review by akoumjian — COMMENTED** [review]
Just a quick read-through. So nice to clean this up!

File: thor/orbits/iod.py (hunk 2 above)
Comment (moeyensj) [reply thor/orbits/iod.py:186]: Yes, good idea. In range and transform we pass the actual class, not a string. Maybe we pass the class all the way through to the worker functions and then initialize it in there?

```python
propagator: Type["Propagator"] = PYOORB,
propagator_kwargs: dict = {},
```

Then inside worker functions:
```python
prop = propagator(**propagator_kwargs)
```

**Comment by akoumjian** [comment]
Yeah, that makes sense to me.

File: thor/orbits/attribution.py (hunk 1 above)
Comment (moeyensj) [reply thor/orbits/attribution.py:74]: No good reason, we were calling generate_ephemeris (original function) with num_jobs=1 which would have called the private function in series. I'll go through and replace the private calls throughout IOD, OD, and the arc extension code.

**Pull request merged**